src/components/Tutorial.vue

```diff
@@ -136,14 +136,14 @@ export default {
   },
   created() {
     this.store.reset();
+    this.store.setReferences(this.references);
```

Comment:
I'm noticing a pattern where you call `setReferences` immediately after the `reset` call. This seems to indicate that you don't really want the "reset" to actually reset the references—in that case, maybe it would be better to just remove the logic for resetting the references in `reset` so that you don't need this additional call following every reset?

Reply:
But the `reset` function, as the name suggests, should reset everything in the store. We are only setting references in the few components that are actually then providing the store. Is it a problem that we call `setReferences` immediately after?

Reply:
It just seemed odd to me that we are basically doing the equivalent of the following with 2 different calls:

```
this.references = {};
this.references = other;
```

Maybe you could just pass the new references to the reset function instead? Or is there ever a case where we want to reset the store and _not_ also update the references?